lib/briefly/entry.ex

```diff
@@ -144,29 +178,43 @@ defmodule Briefly.Entry do
   end
 
   defp generate_tmp_dir(tmp_roots) do
-    {mega, _, _} = :os.timestamp()
-    subdir = "briefly-#{mega}"
+    subdir = path(%{prefix: "briefly-", extname: ""})
 
-    if tmp = Enum.find_value(tmp_roots, &write_tmp_dir(Path.join(&1, subdir))) do
+    if tmp = Enum.find_value(tmp_roots, &write_tmp_dir({&1, subdir})) do
       {:ok, tmp}
     else
       {:error, %Briefly.NoRootDirectoryError{tmp_dirs: tmp_roots}}
     end
   end
 
-  defp write_tmp_dir(path) do
-    case File.mkdir_p(path) do
-      :ok -> path
-      {:error, _} -> nil
+  defp write_tmp_dir({root, path}) do
+    fullpath = Path.join(root, path)
+
+    case File.mkdir_p(root) do
+      {:error, _} ->
+        nil
+
+      :ok ->
+        case File.mkdir(fullpath) do
+          {:error, _} ->
+            nil
+
+          :ok ->
+            case File.chmod(fullpath, Briefly.Config.directory_mode()) do
+              {:error, _} -> nil
+              :ok -> fullpath
+            end
+        end
     end
   end
 
-  defp open(%{directory: true} = options, tmp, attempts, _) when attempts < @max_attempts do
```

Comment:
If we are going to straight up remove this option and not basically convert it to `type: :directory` then we need to make this a major version bump since it's a breaking change. We could alternatively note that it is deprecated in the docs, and then convert `directory: true` to `type: :directory` in the code and do a minor bump for now, with plans for a major release later.

Reply:
I went ahead and added back handling of `directory: true` and added a test that it was still handled. I changed the documentation that I could find all to be `type: :directory`, so that if you want to drop `directory: true` in a major release, you can do that.

Sorry that I missed this.